packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ModelTemplate.java

```diff
@@ -337,6 +314,63 @@ public final void write(ClientModel model, JavaFile javaFile) {
         });
     }
 
+    /**
+     * Initialize the flattened property in flattening properties setter method.
+     * 
+     * <pre>
+     * public ClientModel withFlatteningProperty(FlatteningProperty property) {
+     *     if (this.innerFlattenedProperty() == null) {
+     *         // initializeFlattenedProperty
+     *     }
+     *     this.innerFlattenedProperty().withFlatteningProperty(property);
+     * }
+     * </pre>
+     *
+     * @param methodBlock the setter method block
+     * @param model the client model
+     * @param targetFlattenedProperty flattened property (property.clientFlatten() == true)
+     * @param flatteningProperty flattening property (the property of the flattened property model)
+     * @param streamStyle whether stream-style-serialization is enabled
+     */
+    private static void initializeFlattenedProperty(JavaBlock methodBlock, ClientModel model,
+        ClientModelProperty targetFlattenedProperty, ClientModelPropertyReference flatteningProperty,
+        boolean streamStyle) {
+        String initializeLine;
+        if (streamStyle
+            && model.isAllPolymorphicModelsInSamePackage()
+            && ClientModelUtil.needsPackagePrivateSetter(
+                ClientModelUtil.getDefiningModel(model, targetFlattenedProperty), targetFlattenedProperty,
+                JavaSettings.getInstance(), streamStyle)) {
+            // if all polymorphic models are in the same package, and parent has package-private setter for the
+            // flattened property,
+            // use parent flattened property's setter to initialize
+            initializeLine = String.format("this.%s(new %s());", targetFlattenedProperty.getSetterName(),
+                flatteningProperty.getTargetModelType());
+        } else {
+            initializeLine = String.format("this.%s = new %s();", targetFlattenedProperty.getName(),
+                flatteningProperty.getTargetModelType());
+        }
+        methodBlock.ifBlock(String.format("this.%s() == null", targetFlattenedProperty.getGetterName()),
+            ifBlock -> methodBlock.line(initializeLine));
+    }
+
+    private static JavaVisibility getGetterVisibility(ClientModel model, ClientModelProperty property,
```

Comment:
A better name? I think this only apply to the getter of a flattened property?

Reply:
It applies to all field properties' getters, not only flattened property.
I happen to find that this method visibility applies to public setter as well, which seems not necessary. I'll update the public setter's visibility to always `Public`.

Reply:
OK. My consideration was that "it may apply to all getters, but currently we seems not calling it anywhere else, only here".

Reply:
Enhanced javadoc on this.